We should be able to to just provide a list of POJOs with JAXB annota…

wolfch opened the pull request:
…tions and not have

to provide an ObjectFactory or jaxb.index (which requires ObjectFafctory) or special JAXB
annotated package-info.java.

The use-case is a project with only uses a few JAXB classes and we don't want the whole
JAXB package configuration, especially since JAXBContext.newInstance(...) supports a list
of binding classes.

This change allows configuring JAXBContextFactory with a list of JAXB binding classes.

kdavisk6 commented:
There are few things that will help make this PR ready for review.  Please see [CONTRIBUTING](https://github.com/OpenFeign/feign/blob/master/CONTRIBUTING.md) for more information.

wolfch commented:
Hi, thanks.  I read https://github.com/OpenFeign/feign/blob/master/CONTRIBUTING.md before submitting my PR, so I'm not sure what, exactly, you're referring to.  adriancole suggested a concurrency test, which I have but have not amended to my commit yet.  I will do so shortly.

kdavisk6 commented:
I'm referring to the section about the commit log.  Merge commits make the commit log difficult to read and do not provide information about what was changed and why the merge occurred.  I would suggest either squashing them into one of the other commits or rebasing them with more information about the changes included in the merge.

kdavisk6 commented:
Also, the first line of the PR should comply to the following:
```
First line is a <=72 character description in present tense, explaining what this does.
```

This allows us to discern, at a glance, what the PR is about.

wolfch commented:
Ah, ok.  Those merge commits were just bringing my local up-to-date.  Let me fix that also.  Thanks for the clarification.

wolfch commented:
Closing and re-creating without merge commits shown in log.

wolfch commented:
I removed merge commits and added a concurrent unit test.  But now the 
build fails with :

[ERROR] Failed to execute goal com.mycila:license-maven-plugin:3.0:check 
(default) on project feign-jaxb:

Some files do not have the expected license header -> [Help 1]

Which I did forget to add to the new ConcurrentTest.java, but even after 
adding the copyright comment header and commit --amend/push

it's still complaining - not sure why?


On 3/14/18 7:11 PM, Kevin Davis wrote:
>
> Also, the first line of the PR should comply to the following:
>
> |First line is a <=72 character description in present tense, 
> explaining what this does. |
>
> This allows us to discern, at a glance, what the PR is about.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/OpenFeign/feign/pull/640#issuecomment-373205483>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAh5u6KBwM8dJO5oy_1ixaT4w4Xh-5bIks5teaO7gaJpZM4SBIIH>.
>

-- 
*Chris Wolf*
cw10025@gmail.com <mailto:cw10025@gmail.com>

velo commented:
> it's still complaining - not sure why?

Try: `./mvnw com.mycila:license-maven-plugin:format`.  My best guess is some small typo

wolfch commented:
I re-submitted this PR as a new PR:

https://github.com/OpenFeign/feign/pull/660


On 3/14/18 7:11 PM, Kevin Davis wrote:
>
> Also, the first line of the PR should comply to the following:
>
> |First line is a <=72 character description in present tense, 
> explaining what this does. |
>
> This allows us to discern, at a glance, what the PR is about.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/OpenFeign/feign/pull/640#issuecomment-373205483>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAh5u6KBwM8dJO5oy_1ixaT4w4Xh-5bIks5teaO7gaJpZM4SBIIH>.
>

-- 
*Chris Wolf*
cw10025@gmail.com <mailto:cw10025@gmail.com>
